rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/db/sql/DBSQLFile.kt

```diff
@@ -31,29 +32,45 @@ class DBSQLFile(private val name: String, private val connection: Connection) :
     }
 
     override fun readEqual(key: Value): Record {
-        if (resultSet == null) {
-            chain(emptyList())
+        return readEqual(toFields(key))
+    }
+
+    private fun filterRecord(result: Record): Record {
+        return if (result.matches(lastKey)) {
+            result
+        } else {
+            resultSet?.last()
+            Record()
         }
-        return readFromPositionedResultSet()
     }
 
     override fun readEqual(keys: List<Field>): Record {
-        TODO("not implemented")
+        val result = if (resultSet == null) {
+            chain(emptyList())
+        } else {
+            readFromPositionedResultSet()
+        }
+        lastKey = keys
+        return filterRecord(result)
     }
 
     override fun eof(): Boolean = resultSet?.isLast ?: false
 
     override fun chain(key: Value): Record {
+        return chain(toFields(key))
+    }
+
+    private fun toFields(keyValue: Value): List<Field> {
         val keyName = keys.first()
-        return chain(listOf(Field(keyName, key)))
+        return listOf(Field(keyName, keyValue))
     }
 
     override fun chain(keys: List<Field>): Record {
         val keyNames = keys.map { it.name }
         val sql = "SELECT * FROM $name ${keyNames.whereSQL()} ${keyNames.orderBySQL()}"
```

Comment:
This looks good to me as it is, but if the complexity of the SQL statements increase we could consider using a library or creating a simple internal DSL to build statements

Reply:
Good point. I started using a simple library for creating SQL statements, but it was too heavy at the moment. Let's see how the code evolves.